tests/unit/browser/webkit/network/test_pac.py

```diff
@@ -250,5 +240,37 @@ def serve():
         assert res.fetch_error() is None
     finally:
         serve_thread.join()
+    return res
+
+
+@pytest.mark.skipif(QT_VERSION_STR.startswith('5.7') and
+                    QtWebEngineWidgets is not None and
+                    sys.platform == "linux",
+                    reason="Segfaults when run with QtWebEngine tests on Linux")
+def test_fetch_success():
+    test_str = """
+        function FindProxyForURL(domain, host) {
+            return "DIRECT; PROXY 127.0.0.1:8080; SOCKS 192.168.1.1:4444";
+        }
+    """
+
+    res = fetcher_test(test_str)
     proxies = res.resolve(QNetworkProxyQuery(QUrl("https://example.com/test")))
     assert len(proxies) == 3
+
+
+@pytest.mark.skipif(QT_VERSION_STR.startswith('5.7') and
+                    QtWebEngineWidgets is not None and
+                    sys.platform == "linux",
+                    reason="Segfaults when run with QtWebEngine tests on Linux")
+def test_fetch_evalerror():
+    test_str = """
+        function FindProxyForURL(domain, host) {
+            return "FOO";
+        }
+    """
+
+    res = fetcher_test(test_str)
+    proxies = res.resolve(QNetworkProxyQuery(QUrl("https://example.com/test")))
```

Comment:
You'll need to add a `caplog` argument to this test, and put this line in a  `with caplog.at_level(logging.ERROR):` block - then it won't fail the tests.